### PR TITLE
CSP-1159 Created an api for Get signed up events

### PR DESCRIPTION
**Pull request by ManishAgarwalDfE**

### CODE REVIEW

File: src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryValidator.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using FluentValidation;
+
+namespace SFA.DAS.AANHub.Application.Members.Queries.GetMemberActivities;
+public class GetMemberActivitiesQueryValidator : AbstractValidator<GetMemberActivitiesQuery>
+{
+    public const string MemberIdMissingMessage = "MemberId must have a value";
+
+    public GetMemberActivitiesQueryValidator()
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryValidator.cs:8]: There is already a MemberId validator which should be reused here. https://github.com/SkillsFundingAgency/das-aan-hub-api/blob/main/src/SFA.DAS.AANHub.Application/Common/Validators/MemberId/MemberIdValidator.cs

File: src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesResult.cs
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesResult.cs:1]: The file name should be GetMemberActivitiesQueryResult

File: src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        _auditReadRepository = auditReadRepository;
+    }
+
+    public async Task<ValidatedResponse<GetMemberActivitiesResult>> Handle(GetMemberActivitiesQuery request, CancellationToken cancellationToken)
+    {
+        DateTime rangeStartDate = DateTime.UtcNow.AddMonths(-1 * RangeDuration.EventsRangePeriod);
+        DateTime rangeEndDate = DateTime.UtcNow.AddMonths(RangeDuration.EventsRangePeriod);
+        DateTime currentDate = DateTime.UtcNow;
+
+        var memberActivities = await _attendancesReadRepository.GetAttendances(request.MemberId, rangeStartDate, rangeEndDate, cancellationToken);
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandler.cs:25]: There should be test to assert on the date range that is calculated here.

File: src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        // Assert
+        Assert.Multiple(() =>
+            {
+                Assert.That(result, Is.Not.Null);
+                Assert.That(result.Result, Is.InstanceOf<GetMemberActivitiesResult>());
+                Assert.That(result.Result.LastSignedUpDate, Is.EqualTo(audit.AuditTime));
+                Assert.That(result.Result.EventsPlanned, Is.Not.Null);
+                Assert.That(result.Result.EventsPlanned.Events, Is.Not.Null);
+                Assert.That(result.Result.EventsPlanned.Events.Count, Is.EqualTo(2));
+                Assert.That(result.Result.EventsPlanned.EventsDateRange, Is.Not.Null);
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs:45]: We should be able to predict the date range here

File: src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                Assert.That(result.Result.EventsAttended, Is.Not.Null);
+                Assert.That(result.Result.EventsAttended.Events, Is.Not.Null);
+                Assert.That(result.Result.EventsAttended.Events.Count, Is.EqualTo(3));
+                Assert.That(result.Result.EventsAttended.EventsDateRange, Is.Not.Null);
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs:49]: We should be able to predict the date range here

File: src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            new Attendance(){Id=Guid.NewGuid(),CalendarEventId=Guid.NewGuid(),AddedDate = DateTime.UtcNow.AddDays(-2),MemberId=memberId,IsAttending=true,CalendarEvent = new CalendarEvent(){ Title= "Event Title 2",Urn=null}},
+            new Attendance(){Id=Guid.NewGuid(),CalendarEventId=Guid.NewGuid(),AddedDate = DateTime.UtcNow,MemberId=memberId,IsAttending=true,CalendarEvent = new CalendarEvent(){ Title= "Event Title 3",Urn=null}},
+            new Attendance(){Id=Guid.NewGuid(),CalendarEventId=Guid.NewGuid(),AddedDate = DateTime.UtcNow.AddDays(2),MemberId=memberId,IsAttending=true,CalendarEvent = new CalendarEvent(){ Title= "Event Title 4",Urn=null}},
+            new Attendance(){Id=Guid.NewGuid(),CalendarEventId=Guid.NewGuid(),AddedDate = DateTime.UtcNow.AddDays(5),MemberId=memberId,IsAttending=true,CalendarEvent = new CalendarEvent(){ Title= "Event Title 5",Urn=null}},
+        };
+
+        attendancesReadRepository.Setup(a => a.GetAttendances(It.IsAny<Guid>(), It.IsAny<DateTime>(), It.IsAny<DateTime>(), It.IsAny<CancellationToken>())).ReturnsAsync(attendances);
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs:30]: A specific setup or verify on this call will help

File: src/SFA.DAS.AANHub.Api.UnitTests/Controllers/MembersControllerTests/MembersControllerGetMemberActivitiesTests.cs [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+            Assert.That(memberActivitiesResult.EventsPlanned.EventsDateRange.FromDate, Is.EqualTo(getMemberActivitiesResult.EventsPlanned.EventsDateRange.FromDate));
+            Assert.That(memberActivitiesResult.EventsPlanned.EventsDateRange.ToDate, Is.EqualTo(getMemberActivitiesResult.EventsPlanned.EventsDateRange.ToDate));
+            Assert.That(memberActivitiesResult.EventsPlanned.Events.Count, Is.EqualTo(getMemberActivitiesResult.EventsPlanned.Events.Count));
+            Assert.That(memberActivitiesResult.EventsAttended, Is.Not.Null);
+            Assert.That(memberActivitiesResult.EventsAttended.EventsDateRange, Is.EqualTo(getMemberActivitiesResult.EventsAttended.EventsDateRange));
+            Assert.That(memberActivitiesResult.EventsAttended.EventsDateRange.FromDate, Is.EqualTo(getMemberActivitiesResult.EventsAttended.EventsDateRange.FromDate));
+            Assert.That(memberActivitiesResult.EventsAttended.EventsDateRange.ToDate, Is.EqualTo(getMemberActivitiesResult.EventsAttended.EventsDateRange.ToDate));
+            Assert.That(memberActivitiesResult.EventsAttended.Events.Count, Is.EqualTo(getMemberActivitiesResult.EventsAttended.Events.Count));
+        });
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Api.UnitTests/Controllers/MembersControllerTests/MembersControllerGetMemberActivitiesTests.cs:92]: This is pointless isn't it? All the tests post from top two and last one seems pointless as far as I can see.

**Review by DevenKShah — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryValidator.cs (hunk 1 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryValidator.cs]: Done

File: src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesResult.cs
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesResult.cs:1]: Done

File: src/SFA.DAS.AANHub.Api.UnitTests/Controllers/MembersControllerTests/MembersControllerGetMemberActivitiesTests.cs (hunk 6 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.AANHub.Api.UnitTests/Controllers/MembersControllerTests/MembersControllerGetMemberActivitiesTests.cs]: Done

File: src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs (hunk 5 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs]: Done

File: src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs (hunk 4 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs:53]: Done

File: src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs (hunk 3 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.AANHub.Application.UnitTests/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandlerTests.cs:47]: Done

File: src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandler.cs (hunk 2 above)
Comment (ManishAgarwalDfE) [reply src/SFA.DAS.AANHub.Application/Members/Queries/GetMemberActivities/GetMemberActivitiesQueryHandler.cs:25]: Done

**Review by DevenKShah — DISMISSED** [review]

**Pull request merged**